### PR TITLE
Fix TypeScript type definitions

### LINKED ISSUE #279: Type definition incorrect.
The type definition file dictates that most of the functions provided by the main export return a void value:
https://github.com/honeybadger-io/honeybadger-js/blob/fae5d6178eb08b443dc3a41b1f72f04f050ca58d/honeybadger-js.d.ts#L35
However, it seems this is incorrect upon reviewing the actual source. For instance `configure`:

https://github.com/honeybadger-io/honeybadger-js/blob/fae5d6178eb08b443dc3a41b1f72f04f050ca58d/src/builder.js#L460-L465

### PR DESCRIPTION
**Pull request by joshuap**
Fixes #279

Related: https://github.com/honeybadger-io/honeybadger-react/pull/91

## Status
**READY**

## Description
Fixes a few incorrect return types.

### CODE REVIEW

File: honeybadger-js.d.ts [hunk 1]
```diff
@@ -32,15 +32,15 @@ declare module "honeybadger-js" {
 
     class Honeybadger {
         static apiKey: string;
-        static configure(config: Config): void;
-        static context: any;
+        static configure(config: Config): Honeybadger;
+        static context: object;
```
Comment (schontz) [inline honeybadger-js.d.ts]: `context: any` was right. That allows it to be an object with any properties.

File: honeybadger-js.d.ts [hunk 2]
```diff
@@ -32,15 +32,15 @@ declare module "honeybadger-js" {
...
         static environment: string;
-        static notify(...args: any[]): void;
+        static notify(...args: any[]): object;
```
Comment (schontz) [inline honeybadger-js.d.ts]: Use `any` instead of `object`.

**Comment by joshuap** [comment]
This should take care of that: https://github.com/honeybadger-io/honeybadger-react/pull/91

Assuming that looks right, I'll get both released either later this afternoon or tomorrow morning.

**Comment by schontz** [comment]
Thanks. I’ll install them straightaway so confirm all is well.

**Comment by joshuap** [comment]
@schontz honeybadger-js v2.0.2 and honeybadger-react v0.0.6 have been released with the new type definitions. Let me know how it works for you!

**Comment by schontz** [comment]
@joshuap It works!

I only list `@honeybadger/react` as a dependency. Running `npm i @honeybadger/react@latest` seemed to not pick up the new `honeybadger-js` info, even though it was listed at 2.0.2 in `package-lock.json`. I re-installed the react module and all was well. I think it was just an issue with my TS Server being out-of-date in VS code though. I just mention it in case anyone else has an issue.

Thanks for providing the TypeScript types. It makes integration a lot easier in TS projects.

**Pull request merged**